### PR TITLE
[Fix #10] Bind *1, *2, *3 and *e in cloned session

### LINKED ISSUE #10: *1/*2/*3/*e nil in cloned session
(from https://dev.clojure.org/jira/browse/NREPL-48)

### PR DESCRIPTION
**Pull request by mallt**
- [x] The commits are consistent with our [contribution guidelines](CONTRIBUTING.md)
- [x] You've added tests to cover your change(s)
- [x] All tests are passing
- [ ] The new code is not generating reflection warnings

This PR copies the bindings of *1, *2, *3 and *e to the new session when cloning a session.

### CODE REVIEW

**Comment by bbatsov** [comment]
@mallt Nicely done! Please, mention the bugfix in the changelog as well. 

Seems we need to update our checklist for the PR to mention the changelog. :-)

**Comment by arrdem** [comment]
Nice! LGTM.

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ with the built-in client to an already running nREPL server.
 #### Bugs fixed
 
 * [#38](https://github.com/nrepl/nREPL/issues/38): Remove extra newline in REPL output.
+* [#10](https://github.com/nrepl/nREPL/issues/10): Bind *1, *2, *3 and *e in cloned session.
```
Comment (bbatsov) [inline CHANGELOG.md]: This should go to the `master` section. (you need to to add there "Bugs fixed" section, as there are no entries for `master` yet).

**Comment by bbatsov** [comment]
You might also want to tackle https://github.com/nrepl/nREPL/issues/44 and https://github.com/nrepl/nREPL/issues/33 which are pretty similar in nature. 😉

**Comment by bbatsov** [comment]
@mallt Btw, is the new test passing locally for you on 1.7 and 1.8? For me it fails, but it works on 1.9. Seems there's some degree of flakiness to it.

**Comment by mallt** [comment]
Yes all tests are passing for me locally with `lein test-all`. 

I also noticed #50 but I cannot reproduce this on my machine. I'm using the openjdk java version, maybe that is causing the difference?
```
openjdk version "1.8.0_181"
OpenJDK Runtime Environment (build 1.8.0_181-8u181-b13-0ubuntu0.16.04.1-b13)
OpenJDK 64-Bit Server VM (build 25.181-b13, mixed mode)
```

**Pull request merged**